src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java

```diff
@@ -0,0 +1,461 @@
+                + "      \"current_period_date_formatted\": \"" + CURRENT_PERIOD_FORMATTED + "\",\n"
+                + "      \"called_up_shared_capital_not_paid\": {\n"
+                + "        \"current_amount\": 9,\n"
+                + "        \"previous_amount\": 99\n"
+                + "      },\n"
+                + "      \"previous_period_date_formatted\": \"" + PREVIOUS_PERIOD_FORMATTED
+                + "\",\n"
+                + "    },\n"
+                + "    \"company\": {\n"
+                + "      \"company_number\": \"MYRETON RENEWABLE ENERGY LIMITED\",\n"
+                + "      \"company_name\": \"SC344891\"\n"
+                + "    }\n"
+                + "  }\n"
+                + "}";
+
+        return json;
+    }
+
+    /**
+     * Builds transaction object.
+     *
+     * @return
+     * @throws ParseException
+     */
+    private Transaction createTransaction() throws ParseException {
+        Transaction transaction = new Transaction();
+
+        transaction.setId(TRANSACTION_ID);
+        transaction.setClosedAt(getDateFormatted("Friday, Jul 13, 2018 14:04:56 PM"));
+        transaction.setCompanyNumber(COMPANY_NUMBER);
+        transaction.setKind("transaction");
+        transaction.setStatus("closed");
+        transaction.setUpdatedAt(getDateFormatted("Friday, Jul 13, 2018 14:02:56 PM"));
+        transaction.setLinks(createTransactionLinks());
+
+        return transaction;
+    }
+
+    /**
+     * Builds Account Entity object.
+     *
+     * @return
+     * @throws ParseException
+     */
+    private CompanyAccountEntity createAccountEntity() {
+        CompanyAccountEntity account = new CompanyAccountEntity();
+
+        account.setId(ACCOUNTS_ID);
+
+        CompanyAccountDataEntity accountData = new CompanyAccountDataEntity();
+        accountData.setEtag("etagForTesting");
+        accountData.setKind("accounts");
+        accountData.setLinks(createAccountEntityLinks());
+        accountData.setPeriodEndOn(LocalDate.of(2018, 1, 1));
+        account.setData(accountData);
+
+        return account;
+    }
+
+    /**
+     * Gets the links for the transaction. Self link.
+     *
+     * @return
+     */
+    private Map<String, String> createTransactionLinks() {
+        Map<String, String> transactionLinks = new HashMap<>();
+        transactionLinks.put(LinkType.SELF.getLink(), "/transactions/" + TRANSACTION_ID);
+        return transactionLinks;
+    }
+
+    /**
+     * Creates the links for the accounts data.
+     *
+     * @return
+     */
+    private Map<String, String> createAccountEntityLinks() {
+        Map<String, String> dataLinks = new HashMap<>();
+
+        dataLinks.put(LinkType.SELF.getLink(),
+            String.format("/transactions//%s//company-accounts//%s", TRANSACTION_ID, ACCOUNTS_ID));
+
+        dataLinks.put(LinkType.SMALL_FULL.getLink(),
+            String.format("/transactions//%s//company-accounts//%s//small-full//%s",
```

Comment:
Why are the double slashes needed? Why won't this work:
```java
String.format("/transactions/%s/company-accounts/%s/small-full/%s", TRANSACTION_ID, ACCOUNTS_ID, SMALL_FULL_ID));
```

Reply:
2f26d58